### DISCUSSION TITLE
Is there a way to get all the error messages for a given form generically?

### DISCUSSION THREAD

**Discussion by jim-moody** [react-hook-form/react-hook-form#11698] [Q&A] [answered]
In my example, I have a reusable form that shows a list of error messages and the form can contain 25-30 different inputs with multiple validations so its very tedious to hand-write the path to each possible error message.

Is there a way to just map all the error messages and display them in a list something like this:
https://codesandbox.io/p/sandbox/friendly-williams-2g9c7p?file=%2Fsrc%2FApp.tsx%3A1%2C1-37%2C1

**Comment by Haschikeks** [comment, marked as answer]
Hey,

`error` is a Object. So you could use methods like `Object.values(errors)` to access all of them.

```JSX
<ul>
  {Object.values(errors).map((error) => (
    <li>{error.message}</li>
  ))}
</ul>
```

**Reply by sneko** [reply]
Just note depending on your form it can have a structure of nested objects and nested arrays... so watch out! There is a risk of forgetting listing an error.

Associated discussion: https://github.com/orgs/react-hook-form/discussions/12527

**Reply by kcsujeet** [reply]
> Just note depending on your form it can have a structure of nested objects and nested arrays... so watch out! There is a risk of forgetting listing an error.
> 
> Associated discussion: https://github.com/orgs/react-hook-form/discussions/12527

Is there a way to get all the errors of a particular field? .... Only the first one is returned at the moment.

**Reply by sneko** [reply]
@kcsujeet I don't think so since the message is typed as a string and not an array.
